ludwig/features/image_feature.py

```diff
@@ -166,110 +178,123 @@ def add_feature_data(
                 raise ValueError(
                     'Image height and width must be positive integers'
                 )
-
-        # here if a width and height have not been specified
-        # we assume that all images have the same width and height
-        # thus the width and height of the first one are the same
-        # of all the other ones
-        if (csv_path is None and
-                not os.path.isabs(dataset_df[feature['name']][0])):
-            raise ValueError(
-                'Image file paths must be absolute'
-            )
-
-        first_image = imread(
-            get_abs_path(
-                csv_path,
-                dataset_df[feature['name']][0]
-            )
-        )
-
-        first_img_height = first_image.shape[0]
-        first_img_width = first_image.shape[1]
-        first_img_num_channels = num_channels_in_image(first_image)
-
-        if height == 0 or width == 0:
-            # User hasn't specified height and width
+        else:
+            # User hasn't specified height and width.
+            # So we assume that all images have the same width and height.
+            # Thus the width and height of the first one are the same
+            # as all the other ones
             height = first_img_height
             width = first_img_width
 
-        # User specified num_channels in the model/feature definition
-        user_specified_num_channels = False
-        num_channels = first_img_num_channels
         if NUM_CHANNELS in preprocessing_parameters:
+            # User specified num_channels in the model/feature definition
             user_specified_num_channels = True
             num_channels = preprocessing_parameters[NUM_CHANNELS]
+        else:
+            user_specified_num_channels = False
+            num_channels = first_img_num_channels
 
         assert isinstance(num_channels, int), ValueError(
-            'Number of image channels needs to be an integer')
+            'Number of image channels needs to be an integer'
+        )
+
+        return (should_resize,
```

Comment:
Nit: move should_resize to a new line and closing bracket as well to be consistent with the rest of the codebase.